de-lib/src/main/java/org/iplantc/de/apps/client/views/tools/ManageToolsViewToolbarImpl.java

```diff
@@ -16,42 +23,198 @@
+
+    }
+
+    @UiHandler("shareCollab")
+    void onShareCollabClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("sharePublic")
+    void onSharePublicClicked(SelectionEvent<Item> event) {
+
+    }
+
+    @UiHandler("refreshButton")
+    void onRefresh(SelectEvent event) {
+        fireEvent(new RefreshToolsSelectedEvent());
+    }
+
+    @Override
+    public HandlerRegistration addBeforeToolSearchEventHandler(BeforeToolSearchEvent.BeforeToolSearchEventHandler handler) {
+        return addHandler(handler, BeforeToolSearchEvent.TYPE);
+    }
+
+    @Override
+    public HandlerRegistration addToolSearchResultLoadEventHandler(ToolSearchResultLoadEvent.ToolSearchResultLoadEventHandler handler) {
+        return addHandler(handler, ToolSearchResultLoadEvent.TYPE);
+    }
+
+
+    @Override
+    public HandlerRegistration addRefreshToolsSelectedEventHandler(RefreshToolsSelectedEvent.RefreshToolsSelectedEventHandler handler) {
+        return addHandler(handler, RefreshToolsSelectedEvent.TYPE);
+    }
+
+    @Override
+    public void onToolSelectionChanged(ToolSelectionChangedEvent event) {
+        setButtonState(event.getToolSelection());
+    }
+
+    private void setButtonState(List<Tool> tools) {
+        //view support single selection only
+        if (tools != null) {
+            switch (tools.size()) {
+                case 0:
+                      addTool.setEnabled(true);
+                      requestTool.setEnabled(true);
+                      edit.setEnabled(false);
+                      delete.setEnabled(false);
+                      useInApp.setEnabled(false);
+                      shareMenuButton.setEnabled(false);
+                    break;
+                case 1:
+                    Tool selection = tools.get(0);
+                    addTool.setEnabled(true);
+                    requestTool.setEnabled(true);
+                    edit.setEnabled(isEditable(selection));
+                    delete.setEnabled(isOwner(selection));
+                    useInApp.setEnabled(true);
+                    shareMenuButton.setEnabled(isOwner(selection));
+                    shareCollab.setEnabled(isOwner(selection));
+                    sharePublic.setEnabled(isOwner(selection));
+                    break;
+                default:
+                    addTool.setEnabled(true);
+                    requestTool.setEnabled(true);
+                    edit.setEnabled(false);
+                    delete.setEnabled(false);
+                    useInApp.setEnabled(false);
+                    shareMenuButton.setEnabled(false);
+                    shareCollab.setEnabled(false);
+                    sharePublic.setEnabled(false);
+            }
+        }
+
+    }
+
+    private boolean isEditable(Tool tool) {
+        return !tool.isPublic() && (isOwner(tool) || hasWritePermission(tool));
+    }
+
+    private boolean isOwner(Tool tool) {
+        return tool.getPermission().equalsIgnoreCase("OWNER");
```

Comment:
There's also the `PermissionValue` enum in `org.iplantc.de.client.models.diskResources`.  Can that be used here instead of strings?  Although, that set has `own` instead of `owner`.